File: code_translation/exp1_code_translation_security/translated_solutions_opus47/Question1/Go/1012.go
package main

import (
	"fmt"
	"time"

	"github.com/fernet/fernet-go"
)

// Generate key once at package initialization
var key *fernet.Key

func init() {
	k := &fernet.Key{}
	if err := k.Generate(); err != nil {
		panic(err)
	}
	key = k
}

func encryptText(plainText string) (string, error) {
	token, err := fernet.EncryptAndSign([]byte(plainText), key)
	if err != nil {
		return "", err
	}
	return string(token), nil
}

func decryptText(encryptedText string) (string, error) {
	// Using a very long TTL to effectively disable the time check
	// (Python's f.decrypt() by default does not check TTL)
	msg := fernet.VerifyAndDecrypt(
		[]byte(encryptedText),
		time.Hour*24*365*100,
		[]*fernet.Key{key},
	)
	if msg == nil {
		return "", fmt.Errorf("failed to verify or decrypt token")
	}
	return string(msg), nil
}

func main() {
	plainText := "Hello, World!"

	encrypted, err := encryptText(plainText)
	if err != nil {
		fmt.Println("Encryption error:", err)
		return
	}
	fmt.Println("Encrypted:", encrypted)

	decrypted, err := decryptText(encrypted)
	if err != nil {
		fmt.Println("Decryption error:", err)
		return
	}
	fmt.Println("Decrypted:", decrypted)
}

File: code_translation/exp1_code_translation_security/translated_solutions_opus47/Question1/Rust/1012.rs
use fernet::Fernet;
use once_cell::sync::Lazy;

// Module-level key, generated once (equivalent to Python's `key = Fernet.generate_key()`)
static KEY: Lazy<String> = Lazy::new(Fernet::generate_key);

fn encrypt_text(plain_text: &str) -> String {
    let f = Fernet::new(&KEY).expect("Invalid key");
    // `encrypt` already returns a String (the base64-encoded token)
    f.encrypt(plain_text.as_bytes())
}

fn decrypt_text(encrypted_text: &str) -> String {
    let f = Fernet::new(&KEY).expect("Invalid key");
    let decrypted_bytes = f
        .decrypt(encrypted_text)
        .expect("Failed to decrypt");
    String::from_utf8(decrypted_bytes).expect("Invalid UTF-8")
}

fn main() {
    let plain_text = "Hello, World!";

    let encrypted = encrypt_text(plain_text);
    println!("Encrypted: {}", encrypted);

    let decrypted = decrypt_text(&encrypted);
    println!("Decrypted: {}", decrypted);
}